### PR TITLE
wolfsshd: fix Match User/Group directive misparse

### PR DESCRIPTION
**Pull request by yosuke-wolfssl**
## Summary

`Match User`/`Match Group` directives in `wolfsshd` were misparsed, allowing an authentication-policy bypass.

`AddRestrictedCase()` located the `User`/`Group` keyword in a Match directive value with `XSTRSTR` — an **unanchored substring search**. When a user name contained the substring `Group` (e.g. `GroupAdmin`), the `Group` pass matched *inside* that token and extracted a spurious `groupAppliesTo` value (`Admin`). Symmetrically, a group name containing `User` (e.g. `UserStaff`) produced a ghost `usrAppliesTo`. `wolfSSHD_GetUserConf()` then exact-matched these ghost tokens at every authentication attempt, so any OS account whose name or primary group equaled a ghost token inherited the Match block's overrides (e.g. `PasswordAuthentication yes` overriding a global `no`) — bypassing the intended per-user policy. The reverse direction incorrectly denied access to unrelated users.

## Fix

Replaced the per-keyword substring search with a **positional token-pair parser** (`ParseMatchCriteria`) that walks the value as whitespace-separated tokens:

- A `User`/`Group` keyword is matched as an **exact whole token**.
- The token **immediately following** a keyword is taken literally as the name and is **never re-examined as a keyword** — so a principal named like the opposite keyword (`Match User Group`, a user literally named `Group`) is handled by position, not by substring.
- A recognized keyword with **no following name** (e.g. a bare `Match User`) is now a **configuration error (fail-closed)**, instead of being silently accepted as a dead node that quietly discarded the admin's subsequent settings.
- **Unrecognized** Match criteria are still ignored (kept lenient toward criteria `wolfsshd` does not yet support).

Also:
- Removing `XSTRSTR` eliminates an empty-keyword infinite-loop edge case.
- Fixed a pre-existing leak in `HandleMatch`: the per-match config node was allocated but neither linked nor freed when criteria parsing failed.

## Tests

Added regression tests in `apps/wolfsshd/test/test_configuration.c`:

| Test | Covers |
|------|--------|
| `test_GetUserConfMatchSubstring` | user name containing `Group` → no ghost group token |
| `test_GetUserConfMatchSubstringGroup` | group name containing `User` → no ghost user token |
| `test_GetUserConfMatchLiteralKeywordName` | `Match User Group` → user named `Group`, group unset |
| `test_GetUserConfMatchBareKeyword` | bare `Match User` → fail-closed |
| `test_GetUserConfMatchRepeatedKeyword` | `Match User a User b` → free-then-replace path |
| `test_GetUserConfMatchUserAndGroup` | both criteria on one line → both names extracted |

## Verification

- Full `test_configuration` suite passes.
- Each new test was validated with a **negative control** — it fails against the unfixed/regressed code and passes with the fix.

### CODE REVIEW

File: apps/wolfsshd/configuration.c [hunk 1]
```diff
@@ -871,44 +871,91 @@ static int HandleChrootDir(WOLFSSHD_CONFIG* conf, const char* value)
...
+            while (*pt != '\0' && !IsMatchSeparator(*pt)) {
+                pt++;
+            }
+            tokSz = (int)(pt - tok);
+
+            if (tokSz == 0) {
+                wolfSSH_Log(WS_LOG_ERROR,
+                    "[SSHD] No valid input found with Match");
+                ret = WS_FATAL_ERROR;
+            }
```
[the remark this answers is not in the document]
Comment (yosuke-wolfssl) [reply apps/wolfsshd/configuration.c:942]: Fixed

File: apps/wolfsshd/configuration.c [hunk 2]
```diff
@@ -871,44 +871,91 @@ static int HandleChrootDir(WOLFSSHD_CONFIG* conf, const char* value)
...
 
-/* returns WS_SUCCESS on success, helps with adding a restricted case to the
- * config */
-static int AddRestrictedCase(WOLFSSHD_CONFIG* config, const char* mtch,
-    const char* value, char** out)
+/* returns non-zero if 'c' separates tokens in a Match directive value */
+static int IsMatchSeparator(char c)
+{
+    return (c == ' ' || c == '\t' || c == '\r' || c == '\n');
+}
```
[the remark this answers is not in the document]
Comment (yosuke-wolfssl) [reply apps/wolfsshd/configuration.c]: Fixed

File: apps/wolfsshd/configuration.c [hunk 3]
```diff
@@ -871,44 +871,87 @@ static int HandleChrootDir(WOLFSSHD_CONFIG* conf, const char* value)
 }
 
 
-/* returns WS_SUCCESS on success, helps with adding a restricted case to the
- * config */
-static int AddRestrictedCase(WOLFSSHD_CONFIG* config, const char* mtch,
-    const char* value, char** out)
+/* returns non-zero if 'c' separates tokens in a Match directive value */
```
[the remark this answers is not in the document]
Comment (yosuke-wolfssl) [reply apps/wolfsshd/configuration.c]: Fixed

**Review by ejohnstown — APPROVED** [review]
